src/sdam/events.ts

```diff
@@ -129,14 +166,17 @@ export class TopologyClosedEvent {
  * @public
  * @category Event
  */
-export class ServerHeartbeatStartedEvent {
+export class ServerHeartbeatStartedEvent extends ServerDiscoveryAndMonitoringEvent {
   /** The connection id for the command */
   connectionId: string;
-  /** Is true when using the streaming protocol. */
+  /** Is true when using the streaming protocol */
   awaited: boolean;
+  /** @internal */
+  name = SERVER_HEARTBEAT_STARTED;
 
   /** @internal */
-  constructor(connectionId: string, awaited: boolean) {
+  constructor(connectionId: string, awaited: boolean, topologyId: number) {
```

Comment:
For this and the other heartbeat events I would put topologyId as the first parameter in the constructor, it has a more natural hierarchy: topologyId > connectionId.